### PR TITLE
VirtualMachineSize facts module

### PR DESCRIPTION
**Pull request by nbr23**
##### SUMMARY

Add module to retrieve available [VirtualMachineSize](https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.management.compute.models.virtualmachinesize?view=azure-dotnet) facts.

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

azure_rm_virtualmachinesize_info

### CODE REVIEW

File: plugins/modules/azure_rm_virtualmachinesize_info.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        self.results = dict(
+            changed=False,
+            sizes=[]
+        )
+
+        self.location = None
+        self.name = None
+
+        super(AzureRMVirtualMachineSizeInfo, self).__init__(self.module_arg_spec,
+                                                        supports_check_mode=True,
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_virtualmachinesize_info.py]: lines 121,122,123  continuation line under-indented for visual indent

File: plugins/modules/azure_rm_virtualmachinesize_info.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+                                                        supports_tags=False,
+                                                        facts_module=True)
+
+    def exec_module(self, **kwargs):
+        for key in self.module_arg_spec:
+            setattr(self, key, kwargs[key])
+
+        self.results['sizes'] = self.list_items_by_location()
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_virtualmachinesize_info.py]:
```suggestion
        self.results['sizes'] = self.list_items_by_location()
```

File: plugins/modules/azure_rm_virtualmachinesize_info.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+short_description: Get facts for virtual machine sizes
+
+description:
+    - Get available virtual machine size profiles for a location
+
+options:
+    location:
+        description:
+            - Location for which to list the available virtual machine size profiles
+        required: true
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_virtualmachinesize_info.py:27]:
```suggestion
        required: true
        type: str
```

File: plugins/modules/azure_rm_virtualmachinesize_info.py [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+    name:
+        description:
+            - Name of a size to get information about
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_virtualmachinesize_info.py:31]:
```suggestion
            - Name of a size to get information about
        type: str
```

File: tests/integration/targets/azure_rm_virtualmachinesize_info/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+      that: output['sizes'] | length == 1
+
+- name: Get available sizes for a specific location
+  azure_rm_virtualmachinesize_info:
+    location: "{{ location }}"
+  register: output
+
+- assert:
+      that: output['sizes'] | length > 0
+
```
Comment (Fred-sun) [inline tests/integration/targets/azure_rm_virtualmachinesize_info/tasks/main.yml]:
```suggestion
```

**Review by Fred-sun — COMMENTED** [review]
Small change request!

**Comment by xuzhang3** [comment]
LGTM

**Pull request merged**